src/main/scala/org/constellation/primitives/ChannelMessage.scala

```diff
@@ -1,15 +1,27 @@
 package org.constellation.primitives
 
+import java.util.concurrent.Semaphore
+
+import com.fasterxml.jackson.databind.JsonNode
+import com.github.fge.jsonschema.core.report.ProcessingReport
+import com.github.fge.jsonschema.main.{JsonSchemaFactory, JsonValidator}
 import constellation._
 import org.constellation.DAO
 import org.constellation.util.{MerkleProof, ProductHash, SignatureBatch}
+import org.json4s.jackson.JsonMethods.{asJsonNode, parse}
 
+// Should channelId be associated with a unique keyPair or not?
 case class ChannelMessageData(
                                message: String,
                                previousMessageDataHash: String,
                                channelId: String
                              ) extends ProductHash
 
+case class ChannelOpen(
```

Comment:
arguably better to have some kind of channel trait -- with children like "ValidatedChannel", "BasicChannel" or something. easier to extend to other stuff later as well.

Just an idea though.

Reply:
I think we'll need to store the channels in a better data structure anyways. I'm actually just putting this as JSON on the message field for the first message which is wrong. We should really extend the edge class better so we can make use of TypedEdgeHash and make distinctions for all the different data types.